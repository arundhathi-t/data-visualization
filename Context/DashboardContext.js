// src/context/DashboardContext.js
import React, { createContext, useContext, useState } from 'react';

const DashboardContext = createContext();

export const useDashboardContext = () => {
  return useContext(DashboardContext);
};

export const DashboardProvider = ({ children }) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [timeFilter, setTimeFilter] = useState('all');

  const handleSearchChange = (term) => {
    setSearchTerm(term);
  };

  const handleFilterChange = (filter) => {
    setTimeFilter(filter);
  };

  return (
    <DashboardContext.Provider
      value={{
        searchTerm,
        timeFilter,
        handleSearchChange,
        handleFilterChange,
      }}
    >
      {children}
    </DashboardContext.Provider>
  );
};
