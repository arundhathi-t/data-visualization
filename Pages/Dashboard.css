.dashboard-container {
    font-family: Arial, sans-serif;
    margin: 0 auto;
    max-width: 1200px;
    padding: 20px;
  }
  
  .dashboard-header {
    text-align: center;
    margin-bottom: 20px;
  }
  
  .controls-section {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
    gap: 20px;
  }
  
  .search-bar input {
    padding: 10px;
    width: 300px;
    border: 1px solid #ccc;
    border-radius: 4px;
  }
  
  .time-filters label {
    margin-right: 10px;
    cursor: pointer;
  }
  
  .charts-section {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(400px, 1fr));
    gap: 20px;
  }
  