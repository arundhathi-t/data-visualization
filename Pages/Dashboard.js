// src/pages/Dashboard.js
import React from "react";
import "./Dashboard.css";
import Charts from "../Components/Charts";
import { useDashboardContext } from "../Context/DashboardContext";

const Dashboard = () => {
  const { searchTerm, timeFilter, handleSearchChange, handleFilterChange } =
    useDashboardContext();

  return (
    <div className="dashboard-container">
      <header className="dashboard-header">
        <h1>Data Visualization Dashboard</h1>
      </header>

      <div className="controls-section">
        <div className="search-bar">
          <input
            type="text"
            placeholder="Search metrics..."
            value={searchTerm}
            onChange={(e) => handleSearchChange(e.target.value)}
          />
        </div>

        <div className="time-filters">
          <label>
            <input
              type="radio"
              name="timeFilter"
              value="all"
              checked={timeFilter === "all"}
              onChange={(e) => handleFilterChange(e.target.value)}
            />
            All Time
          </label>
          <label>
            <input
              type="radio"
              name="timeFilter"
              value="lastHour"
              checked={timeFilter === "lastHour"}
              onChange={(e) => handleFilterChange(e.target.value)}
            />
            Last Hour
          </label>
          <label>
            <input
              type="radio"
              name="timeFilter"
              value="lastDay"
              checked={timeFilter === "lastDay"}
              onChange={(e) => handleFilterChange(e.target.value)}
            />
            Last Day
          </label>
          <label>
            <input
              type="radio"
              name="timeFilter"
              value="lastWeek"
              checked={timeFilter === "lastWeek"}
              onChange={(e) => handleFilterChange(e.target.value)}
            />
            Last Week
          </label>
        </div>
      </div>

      <div className="charts-section">
        <Charts searchTerm={searchTerm} timeFilter={timeFilter} />
      </div>
    </div>
  );
};

export default Dashboard;
