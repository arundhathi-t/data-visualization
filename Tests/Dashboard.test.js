// src/tests/Dashboard.test.js
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import { toMatchSnapshot } from "jest-snapshot";
import { DashboardProvider } from "../Context/DashboardContext";
import Dashboard from "../Pages/Dashboard";

// Mock Chart Component (since Charts is a child component)
jest.mock("../components/Charts", () => () => <div>Mocked Charts Component</div>);

describe("Dashboard Component", () => {
  const renderDashboard = () =>
    render(
      <DashboardProvider>
        <Dashboard />
      </DashboardProvider>
    );

  it("renders the dashboard header", () => {
    renderDashboard();
    expect(screen.getByText(/Data Visualization Dashboard/i)).toBeInTheDocument();
  });

  it("renders the search bar and allows input", () => {
    renderDashboard();

    const searchInput = screen.getByPlaceholderText(/Search metrics.../i);
    expect(searchInput).toBeInTheDocument();

    fireEvent.change(searchInput, { target: { value: "test" } });
    expect(searchInput.value).toBe("test");
  });

  it("renders and interacts with time filters", () => {
    renderDashboard();

    const allTimeFilter = screen.getByLabelText(/All Time/i);
    const lastHourFilter = screen.getByLabelText(/Last Hour/i);
    const lastDayFilter = screen.getByLabelText(/Last Day/i);
    const lastWeekFilter = screen.getByLabelText(/Last Week/i);

    expect(allTimeFilter).toBeInTheDocument();
    expect(lastHourFilter).toBeInTheDocument();
    expect(lastDayFilter).toBeInTheDocument();
    expect(lastWeekFilter).toBeInTheDocument();

    // Simulate clicking filters
    fireEvent.click(lastHourFilter);
    expect(lastHourFilter.checked).toBe(true);

    fireEvent.click(lastDayFilter);
    expect(lastDayFilter.checked).toBe(true);

    fireEvent.click(lastWeekFilter);
    expect(lastWeekFilter.checked).toBe(true);
  });

  it("renders the charts section", () => {
    renderDashboard();

    expect(screen.getByText(/Mocked Charts Component/i)).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = renderDashboard();
    expect(asFragment()).toMatchSnapshot();
  });
});
