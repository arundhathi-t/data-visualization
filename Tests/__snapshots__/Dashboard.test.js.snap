// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="dashboard-container"
  >
    <header
      class="dashboard-header"
    >
      <h1>
        Data Visualization Dashboard
      </h1>
    </header>
    <div
      class="controls-section"
    >
      <div
        class="search-bar"
      >
        <input
          placeholder="Search metrics..."
          type="text"
          value=""
        />
      </div>
      <div
        class="time-filters"
      >
        <label>
          <input
            checked=""
            name="timeFilter"
            type="radio"
            value="all"
          />
          All Time
        </label>
        <label>
          <input
            name="timeFilter"
            type="radio"
            value="lastHour"
          />
          Last Hour
        </label>
        <label>
          <input
            name="timeFilter"
            type="radio"
            value="lastDay"
          />
          Last Day
        </label>
        <label>
          <input
            name="timeFilter"
            type="radio"
            value="lastWeek"
          />
          Last Week
        </label>
      </div>
    </div>
    <div
      class="charts-section"
    >
      <div>
        Mocked Charts Component
      </div>
    </div>
  </div>
</DocumentFragment>
`;
